### DISCUSSION TITLE
router.push() with query doenst work correctly for some reasons. help me!

### DISCUSSION THREAD

**Discussion by tofuwu-dev** [vercel/next.js#49295] [Help] [answered]
### Summary

Hi @everybody

I have the following link: http://localhost:3000/swap?from=from&to=to

And I am gonna replace from and to. so I want to link like this: http://localhost:3000/swap?from=to&to=from

For it, I did some work as is:
```bash
const changeLink = () => {
   const { from, to } = router.query;
   router.push(`/swap?from=${to}&to=${from}`);
}
```

so when I click button, it calls `changeLink` function.

But it didnt work as I expected. It redirects like this - http://localhost:3000/swap?from=to
but where is `&to=from`

Any ideas? Thanks in advance!

**Comment by icyJoseph** [comment]
Hi,

It works for me... what version of next.js are you using? - is it app dir? or pages dir?

**Reply by tofuwu-dev** [reply]
pages dir

**Reply by icyJoseph** [reply, marked as answer]
Yeah it works well. https://stackblitz.com/edit/nextjs-ey4ibz?file=pages%2Fswap.js,pages%2Findex.js

**Reply by tofuwu-dev** [reply]
Hi @icyJoseph 

Glad to receive your answer. I figured out the issue and it occured from my bad. I could find the solution soon.
But I am very grateful for your very quick reply! I will mark this as answer.

**Reply by icyJoseph** [reply]
Nice! I guess it was something with _app? or something else going on the code? a use-effect? Glad it was fixed!

**Reply by tofuwu-dev** [reply]
use-effect. :)
